staging/src/k8s.io/apiserver/pkg/endpoints/filters/trace.go

```diff
@@ -56,8 +57,16 @@ func WithTrace(handler http.Handler, longRunningCheck genericapirequest.LongRunn
 			utiltrace.Field{Key: "user-agent", Value: &internal.LazyTruncatedUserAgent{req}},
 			utiltrace.Field{Key: "client", Value: &internal.LazyClientIP{req}})
 		req = req.Clone(ctx)
-		// Set trace as root trace in context for nested tracing
-		defer trace.LogIfLong(30 * time.Second)
+		defer func(start time.Time) {
+			if time.Since(start) >= 30*time.Second {
```

Comment:
I'm not sure I follow this, is there like both a step trigger and an overall trigger or something?